### PR TITLE
feat(segment): implement iOS 13 segment with animation

### LINKED ISSUE #18663: feat: iOS 13 Segmented Control
# Feature Request

**Ionic version:**

[x] **4.x**

**Describe the Feature Request**

Update the iOS Segment to the iOS 13 design.

**Describe Preferred Solution**

58835788-bd49bf80-860b-11e9-921c-c0bfd39e3511
58835791-bf138300-860b-11e9-91b4-2fee6dc04a3c

### PR DESCRIPTION
**Pull request by brandyscarney**
:mag: READY FOR REVIEW :mag: 

Docs PR: https://github.com/ionic-team/ionic-docs/pull/1072

## Dev Build

`5.0.0-dev.201911072022.7909b12`

## Changes

Closes #18663

- Converts Segment to shadow
- Enables gesture to swipe between segment buttons
- Adds indicator transition to slide the indicator between buttons
- Updates global theme variables 
- Removes activated state, now handled by the gesture
- Updates iOS to latest iOS 13 UI
- Ensures customization is working for the buttons and indicator
- Updates the e2e tests
- Full list of changes: https://gist.github.com/brandyscarney/e6cfe43c359bb2c932e12f8d615e1669

### iOS

| old                   | new                   |
| ----------------------| ----------------------|
| old basic | new basic |
| old colors | new colors |
| old custom | new custom |
| old spec | new spec |
| old toolbar | new toolbar |

### Material Design

| old                   | new                   |
| ----------------------| ----------------------|
| old basic| new basic |
| old colors| new colors |
| old custom| new custom |
| old spec| new spec |
| old toolbar| new toolbar|

### CODE REVIEW

**Comment by brandyscarney** [comment]
Current dev build: `5.0.0-dev.201910312304.0faefdb`

File: core/src/components/segment/segment.tsx [hunk 1]
```diff
@@ -78,11 +89,261 @@ export class Segment implements ComponentInterface {
...
+
+    if (!indicator) {
+      return;
+    }
+
+    const reduceMotion = window.matchMedia('(prefers-reduced-motion: reduce)').matches;
+
+    if (activated && mode === 'ios' && !reduceMotion) {
+      indicator.style.setProperty('transform', 'scale(0.95)');
+    } else {
```
Comment (liamdebeasi) [inline core/src/components/segment/segment.tsx]: Are we able to do this from the CSS? Maybe use a css var or something:

`indicator.style.setProperty('transform', 'var(--indicator-scale)');`

File: core/src/components/segment/segment.tsx [hunk 2]
```diff
@@ -78,11 +89,261 @@ export class Segment implements ComponentInterface {
...
+
+    detail.event.preventDefault();
+    detail.event.stopImmediatePropagation();
+
+    const ripple = (detail.event!.target! as HTMLElement).shadowRoot!.querySelector('ion-ripple-effect');
+    if (!ripple) { return; }
+
+    const { x, y } = pointerCoord(detail.event);
+
+    ripple.addRipple(x, y).then(remove => remove());
```
Comment (liamdebeasi) [inline core/src/components/segment/segment.tsx]: So I'm actually trying this on an Android device and the ripple is being applied to the starting element not the ending element.

Steps to repro:

1. Load the Basic test on Android.
2. Start on "Free" and and swipe to "Top".
3. Release finger. Notice that the ripple is applied to "Free" and not "Top".

File: core/src/components/segment/segment.tsx (hunk 1 above)
Comment (brandyscarney) [reply core/src/components/segment/segment.tsx]: Doing this now using CSS only

**Pull request merged**